packages/cache/README.md

```diff
@@ -1,3 +1,527 @@
 # @data-eden/cache
 
-TODO: writeme
+## Principles
+
+- **Configurable**: Configuration to allow for application wide default
+  settings as well as per-query settings (e.g. handling non-standard API
+  endpoints, opting out of [cached entity](#definitions) merging per query).
+- **Extensible & Layered**: Supplies extension points allowing custom caching
+  implementations (e.g. persistent caching via IndexDB). Users are _not_ limited to
+  out of the box caching configurations.
+- **Zero Config**: Works with sensible default settings "out of the box" with
+  minimal user setup.
+- **Independent**: Caching system is not "aware" of GraphQL semantics or the
+  specifics of how data is loaded (e.g. `fetch` layer, websockets, etc).
+- **Ergonomic Invalidation**: Cache invalidation is notoriously difficult. This
+  system attempts to make invalidation as easy as possible. APIs are designed
+  up front to enable the cache to absorb the complexities of invalidation (and
+  _not_ force that complexity onto the user).
+- **Debuggable**: Exposes the ability to debug caching details. Understanding
+  the sources of the _contents_ of the cache (e.g. [entities](#definitions) being merged as a
+  result of multiple queries) or the reason specific entities are being
+  retained due to caching configuration should not require deep understanding
+  of the cache internals.
+- **Consistent**: A specific [cached entities](#definitions) Identical cached entities loaded across multiple queries are
+  updated "live". Extension points exist to trigger rerendering for various
+  frameworks (e.g. Ember, Glimmer, React, etc).
+
+## Features
```

Comment:
Probably this use case is obvious through some other term in the notes here but I think we should explicitly call out the "cache consistency" usecase that applications are used to today "ensuring that changes you make in one part of your UI are propagated to other parts of the UI. " 

I see our implementation will have an identity map with indexedDB, but do we need to call out in implementation notes or deep dive on how the new cache system will keep records in sync?

Reply:
> I think we should explicitly call out the "cache consistency" usecase

+1 good callout.


> I see our implementation will have an identity map with indexedDB

To clarify: you mean you see how our implementation *could* use IndexedDB?  I wouldn't expect IndexedDB use to be required or necessarily configured by default out of the box.

> but do we need to call out in implementation notes or deep dive on how the new cache system will keep records in sync?

Can you say more here?  I don't fully follow this question. Is this about merging (i.e. cache consistency) or are you referring to some other synchronization?

Reply:
For indexedDB, I was looking at this example note `e.g. a get hook that falls back to reading from IndexedDB or make a network call)` from the implementation notes and got confused, yeah I see we don't need that. The mapping that we are creating handles the storage. 

For cache-consistency implementation, to put it in a simple way, I was trying to imagine how the sync would work. I was not sure if merging would take care of that or if anything else is needed to maintain consistency.